lib/maths/CTimeSeriesDecompositionDetail.cc

```diff
@@ -1317,9 +1324,12 @@ void CTimeSeriesDecompositionDetail::CComponents::handle(const SDetectedSeasonal
         const CPeriodicityHypothesisTestsResult& result{message.s_Result};
         const CExpandingWindow& window{message.s_Window};
         const TPredictor& predictor{message.s_Predictor};
+        const TModelChangeCallback& onModelChange{message.s_ModelChangeCallback};
 
         if (this->addSeasonalComponents(result, window, predictor)) {
-            LOG_DEBUG(<< "Detected seasonal components at " << time);
+            std::string annotation = "Detected seasonal components";
+            LOG_DEBUG(<< annotation << " at " << std::to_string(time));
+            onModelChange(time, annotation);
```

Comment:
I think this isn't the right place to add the annotation and it would be nice to include information about the periodic component detected. Let's discuss this a bit offline.

Reply:
Done.

I've moved reporting the annotation to `addSeasonalComponents` method.
I've also added reporting in `addCalendarComponent` method